manifests/ingress.yaml

```diff
@@ -0,0 +1,17 @@
+# kubectl apply -f ./manifests/ingress.yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: api-gateway-rules
+  namespace: default
+  annotations:
+    kubernetes.io/ingress.class: nginx
+spec:
+  rules:
+    - http:
+        paths:
+          # Only expose the Dapr invoke API, lets us call our services and nothing more
+          - path: /
```

Comment:
After thinking about it a bit, exposing all dapr services to the open internet feels too much. I think that I would consider the HR API adapter to be a private service that should only be accessed via the slack bot or something?

I think that the minimal thing we need to expose to the open internet is the /ping endpoint of the slack service. Exposing that via dapr might be enough to demonstrate the usefulness of dapr? The slack service could then call the hr api via dapr when we add a /whosmymanager command, to tie everything together in a later pr.

Reply:
I think the plan is to have the slack service in `/lib` access the HR API, the slack service should be exposed publicly but the HR service should only accept internal cluster communication.
We might want to add istio (unless dapr can do something similar) so we can easily control internal vs external traffic.

Reply:
Hi both. Istio is quite a heavyweight thing (as is Dapr), and the Dapr documentation definitely talks about its advantages over Istio. Based on my (definitely quite beginnerish) knowledge, it seems like having both is probably not the best way forward.

One thing to note is that I found it quite difficult setting things up so that Dapr could be invoked from outside the cluster, and this pull request is meant to embody what I learnt (and basically nothing else). 

As yet I haven't tried to Dapr-ize the HR api, and I think the Slack service is also not Dapr-ized, so I would probably propose leaving this as it is for the time being (possibly using the Dapr security system is the usual thing for example, as opposed to locking down the ingress).

Perhaps we could add some comments here to indicate that it is a temporary thing?

Reply:
I've added some comments to indicate that the ingress rules are a temporary measure